scripts/iscsictl.py

```diff
@@ -155,10 +155,16 @@ def connect(self):
         if not self._copy_id:
             self.ssh_copy_id()
 
-        self._connect = sh.ssh.bake('-i', self.key.key(),
-                                    '-o', 'StrictHostKeyChecking=no',
-                                    '-o', 'UserKnownHostsFile=/dev/null',
-                                    '%s@%s' % (self.user, self.host))
+        if self.key:
+            self._connect = sh.ssh.bake('-i', self.key.key(),
+                                        '-o', 'StrictHostKeyChecking=no',
+                                        '-o', 'UserKnownHostsFile=/dev/null',
+                                        '%s@%s' % (self.user, self.host))
+        else:
+            self._connect = sh.ssh.bake('-o', 'StrictHostKeyChecking=no',
```

Comment:
Actually this is not code duplication, because the duplication that you see is in the data.  I find this more explicit (`PEP 20`) than the second best option (refactor the parameters and expand them with `*`)

But I will use the second option.